src/components/App.tsx

```diff
@@ -394,7 +393,10 @@ class App extends React.Component<AppProps, AppState> {
         }
       >
         <ExcalidrawContainerContext.Provider
-          value={this.excalidrawContainerRef.current}
+          value={{
+            container: this.excalidrawContainerRef.current,
+            id: this.id,
+          }}
```

Comment:
this will mess up referential integrity, thus rerender consumers unnecessarily, am I correct? And it looks like most of the UI is subscribing to this context.

If so, I'd suggest either 1) splitting into two context providers, or 2) factor the object out and update it only if either of the two values change.

Reply:
> this will mess up referential integrity, thus rerender consumers unnecessarily, am I correct? And it looks like most of the UI is subscribing to this context.

When do you think that will happen? The `id` is initialized only once in the constructor and would remain the same throughout the lifecycle of the comp...

Reply:
I meant the value object `{}` the attributes are in (a new object will be created on every render). `Context.Provider` should be doing a simple equality check: https://reactjs.org/docs/context.html#contextprovider